### PR TITLE
docs: add changelog and complete documentation

### PR DESCRIPTION
**Pull request by JuanDavidBuitrago**
Description:
This PR adds changelog and repository documentation.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+* Add requirements files.
+* Add Add required maintenance files.
+
+[12.0.2] - 2022-04-29
+~~~~~~~~~~~~~~~~~~~~
+
+Added
+_____
+
+* First release on PyPI.
```
Comment (mariajgrimaldi) [inline CHANGELOG.rst]: This is not true. Maybe first git tagged release, but not on pypi.

File: README.rst [hunk 2]
```diff
@@ -1,6 +1,11 @@
 codejail plugin for `Tutor <https://docs.tutor.overhang.io>`__
 ===================================================================================
 
+The repository Tutor contrib codejail, is a codejail plugin to manages execution of
+untrusted code in secure sandboxes. It is designed primarily for Python execution, but
+can be used for other languages as well. Security is enforced with AppArmor.
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
Tutor plugin that enables execution of untrusted code in secure sandboxes using an external [service](https://github.com/eduNEXT/codejailservice) based on the [codejail](https://github.com/openedx/codejail) library.
```
Most of what you described here is part of codejail itself, and this is just a tutor plugin that helps with tutor + codejail.

File: README.rst [hunk 3]
```diff
@@ -24,6 +29,75 @@ Finally, the platform can be run as usual::
...
+|       Maple       |    >= 13.x     |
++-------------------+----------------+
+
+Functionality test
+------------------
+
+How to know if codejail is working
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The easiest way to test if codejail is working is to validate it in ``Studio`` with a course
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
The easiest way to test whether codejail is working is to validate it in ``Studio`` with a course
```

File: README.rst [hunk 4]
```diff
@@ -24,6 +29,75 @@ Finally, the platform can be run as usual::
...
+that has loncapa problems exercises.
+
+This test was performed on the Maple version of Open Edx, using the course ``course_codejail_example.tar.gz``
```
Comment (mariajgrimaldi) [inline README.rst]: can you link the course?

Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
This test was performed on the Maple version of Open edX, using the course ``course_codejail_example.tar.gz``
```

File: README.rst [hunk 5]
```diff
@@ -24,6 +29,75 @@ Finally, the platform can be run as usual::
...
+found in the additional resources section.
+
+Once the course is imported, enter any section and select an exercise, the result that we must obtain in the
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
Once the course is imported, go to any section and select an exercise, the proper result is:
```

File: README.rst [hunk 6]
```diff
@@ -24,6 +29,75 @@ Finally, the platform can be run as usual::
...
+    cannot create LoncapaProblem block-v1:edX+DemoX+Demo_Course+type@problem+block@integral1: Error while
+    executing script code: Codejail API Service is unavailable. Please try again in a few minutes.
+
+.. image:: ./resourcesfiles/Codejailfail.png
+    :width: 750px
+    :align: center
+    :alt: Example when codejail is not working
+
+This indicates that the codejail service is not turned on or is not working properly. Be sure to follow the
+steps in Usage so this doesn't happen.
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
steps in the usage section so this doesn't happen.
```
And please add the reference to the section

File: README.rst [hunk 7]
```diff
@@ -24,6 +29,75 @@ Finally, the platform can be run as usual::
...
+Once the course is imported, enter any section and select an exercise, the result that we must obtain in the
+example course is the following:
+
+.. image:: ./resourcesfiles/Codejailworking.png
+    :width: 725px
+    :align: center
+    :alt: Example when codejail is working
+
+Indeed, the content of the course in the section will be presented correctly, without any type of errors and can
+be verified by doing what is found in the instructions of the course.
```
Comment (mariajgrimaldi) [inline README.rst]:
```suggestion
In this case, the section's content will render correctly and will be working as specified in the instructions of the problem.
```

File: README.rst (hunk 5 above)
Comment (mariajgrimaldi) [reply README.rst]: any section uses codejail?

File: docs/how_tos/CONTRIBUTING.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
+################################
```
Comment (mariajgrimaldi) [inline docs/how_tos/CONTRIBUTING.rst]: I thought this file was supposed to be in the root folder

File: README.rst [hunk 9]
```diff
@@ -1,6 +1,11 @@
 codejail plugin for `Tutor <https://docs.tutor.overhang.io>`__
```
Comment (mariajgrimaldi) [inline README.rst:1]: Please, add the type of language to the bash instructions!

File: README.rst (hunk 4 above)
Comment (JuanDavidBuitrago) [reply README.rst]: Of course

File: README.rst (hunk 5 above)
Comment (JuanDavidBuitrago) [reply README.rst]: All sections use codejail.

File: docs/how_tos/CONTRIBUTING.rst (hunk 8 above)
Comment (JuanDavidBuitrago) [reply docs/how_tos/CONTRIBUTING.rst]: Changed to root folder

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 10]
```diff
@@ -0,0 +1,26 @@
+1. Purpose of this Repo
```
Comment (mariajgrimaldi) [inline docs/decisions/0001-purpose-of-this-repo.rst]: This file should indicate _why_ we are creating this repository, and I haven't found that reason here. 

1. Context: explain why a plugin for codejail was needed
2. Decision: create a tutor-contrib-plugin that... explain what this plugin does
3. Consequences: consequences of the creation of this plugin (affecting directly how codejail is used)

**Comment by mariajgrimaldi** [comment]
Please, resolve comments _after_ pushing the changes.

File: docs/decisions/0001-purpose-of-this-repo.rst [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+--------
+
+Create a tutor-contrib-plugin that facilitates the installation of the codejail service, using tutor commands
+that pull Docker images and create the container where the service runs, independent of the platform's own containers.
+
+Consequences
+------------
+
+The creation of this plugin affects the installation and operation of the codejail service, adapting the necessary instances
+so that it works according to the service specifications.
```
Comment (mariajgrimaldi) [inline docs/decisions/0001-purpose-of-this-repo.rst:25]: Good!

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**